### ISSUE TITLE
Flatens file structure - subfolders contents go to root

### ISSUE THREAD

**Issue by OzBob** [LanceMcCarthy/Action-AzureBlobUpload#124]
This setup is flattening the file structure in the Blob destination

code:
````
      - name: Azure Blob Upload with Destination folder
      uses: LanceMcCarthy/Action-AzureBlobUpload@v1.8.4
      with:
        connection_string: ${{ secrets.AZBLOBCONNECTIONSTRING }}
        container_name: '$web'
        source_folder: dist/i6
        destination_folder: i6
        clean_destination_folder: true
````

Log says:
```
Uploaded dist/i6/assets/svg-country-flags/svg/bi.svg to $web/i6/bi.svg
```

This:

        source_folder: dist/i6
also flattens.

Temp work around, each folder has a build step, e.g I have now added:

```

      - name: Blob 1
      uses: LanceMcCarthy/Action-AzureBlobUpload@v1.8.4
      with:
        connection_string: ${{ secrets.AZBLOBCONNECTIONSTRING }}
        container_name: '$web'
        source_folder: dist/i6
        destination_folder: i6
        clean_destination_folder: true
        is_recursive: **false**

      - name: Blob 2
      uses: LanceMcCarthy/Action-AzureBlobUpload@v1.8.4
      with:
        connection_string: ${{ secrets.AZBLOBCONNECTIONSTRING }}
        container_name: '$web'
        source_folder: dist/i6/assets
        destination_folder: i6/assets
        clean_destination_folder: **false**
        is_recursive: **false**
```

**Comment by OzBob** [comment]
This:
```
        source_folder: dist/i6
```
also flattens.

**Comment by LanceMcCarthy** [comment]
Thanks for sharing. This indicates a problem in the recently added helper methods that fixed another issue. @danielgary did you notice anything like this?

I'm playing whackamole with these path issues, the past 6 or 7 releases have been to address new situations that the tests have missed.

**Comment by LanceMcCarthy** [comment]
I found the cause of this and will be pushing a fix before the end of the day.

**Comment by LanceMcCarthy** [comment]
I've [published v1.9.0](https://github.com/LanceMcCarthy/Action-AzureBlobUpload/releases/tag/v1.9.0) which includes a fix for this issue. Thank you for your patience and understanding.

Let me know if you have any further issues.

**Comment by OzBob** [comment]
sorry, 1.9.0 still flattens the subfolders into the root
